server/schema/order/res.js

```diff
@@ -0,0 +1,81 @@
+import {Order, OrderDetail, Menu, Modifier, sequelize} from '../../models'
+import {_auth} from '../../util'
+
+/**
+ * @param {*} orderDetail
+ * {
+    orderId: Int
+    menuId: Int
+    modifierIds: [Int]
+    quantity: Int
+    price: Float
+ * }
+ */
+async function calcOrderDetailPrice(orderDetail) {
+  let menuPrice = 0,
+    modifiersPrice = 0
+  try {
+    menuPrice = await fetchMenuPrice(orderDetail.menuId)
+    modifiersPrice = await fetchModifersPrice(orderDetail.modifierIds)
+    return (menuPrice + modifiersPrice) * orderDetail.quantity
+  } catch (error) {
+    throw new Error(error.message)
+  }
+}
+
+async function fetchMenuPrice(menuId) {
+  try {
+    let price = await Menu.findOne({where: {id: menuId}}).get('price')
+    return parseFloat(price)
+  } catch (error) {
+    throw new Error(error.message)
+  }
+}
+async function fetchModifersPrice(modifierIds) {
+  try {
+    let prices = 0
+    await Promise.all(
+      modifierIds.map(async modifierId => {
```

Comment:
hit db too many times, pls hit 1 time only to get all the modifiers back, then do the calculation for the returned Array of Modifiers by using lodash sumBy -> https://lodash.com/docs#sumBy